### PR TITLE
Add documentation for building tests against packages

### PR DESCRIPTION
**Pull request by chcosta**

### CODE REVIEW

**Comment by chcosta** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test

File: Documentation/building/build-tests-against-packages.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## Usage Scenarios
+
+### Build the product locally and then test what you've built
+
+In this scenario, you produce a product build, including packages, then generate test project.json's which reference the locally built package versions and finally build tests compiling against the local packages.
+
+1. Build the product and packages 
+ - ```Build.cmd -BuildTests=false```
+2. Generate Test project.json files
+ - ```msbuild /p:BuildTestsAgainstPackages=true /p:SkipCleanPackages=true /p:PackagesDrops=[ProjectDir]bin/packages/[Relase|Debug]/ /t:BatchGenerateTestProjectJsons /t:BatchRestorePackages```
```
Comment (weshaggard) [inline Documentation/building/build-tests-against-packages.md]: Why not a call to sync? I think most of that parameters are part of sync -t

File: Documentation/building/build-tests-against-packages.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  - /p:SkipCleanPackages=true is required for release branches where the packages folder is cleaned during every build.
+3. Build Tests against packages
+ - ```build-tests.cmd -BuildTestsAgainstPackages -- /p:"PackagesDrops=[ProjectDir]bin/packages/[Relase|Debug]/"```
```
Comment (weshaggard) [inline Documentation/building/build-tests-against-packages.md]: Relase -> Release (this typo is in a few places).

File: Documentation/building/build-tests-against-packages.md (hunk 1 above)
Comment (weshaggard) [inline Documentation/building/build-tests-against-packages.md]: What does SkipCleanPackages do exactly? We aren't trying to do a clean here.

Comment (chcosta) [inline Documentation/building/build-tests-against-packages.md]: https://github.com/dotnet/corefx/commit/909d621a7f067dd63e225399ac11344e1c84e51e

File: Documentation/building/build-tests-against-packages.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+This scenario uses a versions file (https://github.com/dotnet/versions/blob/master/build-info/dotnet/corefx/master/Latest_Packages.txt, for example) to determine what package versions to build tests against.
+
+1. Generate Test project.json files using a 'versions' file.
+ -   ```msbuild /p:BuildTestsAgainstPackages=true /p:SkipCleanPackages=true /p:VersionsFiles=[local version file path] /t:BatchGenerateTestProjectJsons /t:BatchRestorePackages```
+  - /p:SkipCleanPackages=true is required for release branches where the packages folder is cleaned during every build.
+2. Build Tests against packages
+ - ```build-tests.cmd -BuildTestsAgainstPackages -- /p:"PackagesDrops=[ProjectDir]bin/packages/[Relase|Debug]/"```
+  - -BuildTestsAgainstPackages tells the build to use the project.json files you generated in the "Generate Test project.json files" step
+  - /p:"PackagesDrops=[ProjectDir]bin/packages/[Release|Debug]/" tells the build to use the packages from your local build drop.
```
Comment (weshaggard) [inline Documentation/building/build-tests-against-packages.md]: If you are using the versions file wouldn't you be downloading all the packages from the feed? If so why do you need the PackagesDrops parameter?

**Pull request merged**